metadata-ingestion/src/datahub/ingestion/source/powerbi_report_server/report_server.py

```diff
@@ -0,0 +1,728 @@
+        """
+        LOGGER.info("Converting user {} to datahub's user".format(user.username))
+
+        # Create an URN for User
+        user_urn = builder.make_user_urn(user.get_urn_part())
+
+        user_info_instance = CorpUserInfoClass(
+            displayName=user.properties.display_name,
+            email=user.properties.email,
+            title=user.properties.title,
+            active=True,
+        )
+
+        info_mcp = self.new_mcp(
+            entity_type=Constant.CORP_USER,
+            entity_urn=user_urn,
+            aspect_name=Constant.CORP_USER_INFO,
+            aspect=user_info_instance,
+        )
+
+        # removed status mcp
+        status_mcp = self.new_mcp(
+            entity_type=Constant.CORP_USER,
+            entity_urn=user_urn,
+            aspect_name=Constant.STATUS,
+            aspect=StatusClass(removed=False),
+        )
+
+        user_key = CorpUserKeyClass(username=user.username)
+
+        user_key_mcp = self.new_mcp(
+            entity_type=Constant.CORP_USER,
+            entity_urn=user_urn,
+            aspect_name=Constant.CORP_USER_KEY,
+            aspect=user_key,
+        )
+
+        return [info_mcp, status_mcp, user_key_mcp]
+
+    def to_datahub_work_units(self, report: Report) -> Set[EquableMetadataWorkUnit]:
+        mcps = []
+
+        LOGGER.info("Converting Dashboard={} to DataHub Dashboard".format(report.name))
+        # Convert user to CorpUser
+        user_mcps = self.to_datahub_user(report.user_info)
+        # Convert tiles to charts
+        ds_mcps: List[Any]
+        chart_mcps: List[Any]
+        # ds_mcps = self.__to_datahub_dataset(report)
+        chart_mcps, ds_mcps = [], []  # self.to_datahub_chart(dashboard.tiles)
+        # Lets convert Dashboard to DataHub Dashboard
+        dashboard_mcps = self.__to_datahub_dashboard(report, chart_mcps, user_mcps)
+
+        # Now add MCPs in sequence
+        mcps.extend(ds_mcps)
+        mcps.extend(user_mcps)
+        mcps.extend(chart_mcps)
+        mcps.extend(dashboard_mcps)
+
+        # Convert MCP to work_units
+        work_units = map(self.__to_work_unit, mcps)
+        # Return set of work_unit
+        return OrderedSet([wu for wu in work_units if wu is not None])
+
+
+@dataclass
+class PowerBiReportServerDashboardSourceReport(SourceReport):
+    scanned_report: int = 0
+    filtered_reports: List[str] = dataclass_field(default_factory=list)
+
+    def report_scanned(self, count: int = 1) -> None:
+        self.scanned_report += count
+
+    def report_dropped(self, view: str) -> None:
+        self.filtered_reports.append(view)
+
+
+@platform_name("PowerBIReportServer")
+@config_class(PowerBiReportServerDashboardSourceConfig)
+@support_status(SupportStatus.UNKNOWN)
+@capability(SourceCapability.OWNERSHIP, "Enabled by default")
+class PowerBiReportServerDashboardSource(Source):
+    """
+        This plugin extracts the following:
+
+    - Power BI Dashboards, tiles, datasets
```

Comment:
Please update comments as per concept mapping.

Reply:
update comments to powerbi report server module